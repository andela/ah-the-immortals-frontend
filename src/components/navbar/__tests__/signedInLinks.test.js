import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import toJson from 'enzyme-to-json';
import SignedInLinks from '../signedInLinks';

describe('Tests rendering of links on navigation bar when the user is logged in', () => {
  it('renders signed in links component', () => {
    const history = createBrowserHistory();
    const wrapper = mount(
      <Router history={history}>
        <SignedInLinks />
      </Router>

    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });   
});
