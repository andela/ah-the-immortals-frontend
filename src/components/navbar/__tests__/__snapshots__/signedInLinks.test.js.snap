// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests rendering of links on navigation bar when the user is logged in renders signed in links component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SignedInLinks
    handleLogout={[Function]}
    username=""
  >
    <div
      className="nav ml-auto"
    >
      <NavLink
        className="nav-link mt-1 mr-2"
        to="/postarticle"
      >
        <Link
          aria-current={null}
          className="nav-link mt-1 mr-2"
          to="/postarticle"
        >
          <a
            aria-current={null}
            className="nav-link mt-1 mr-2"
            href="/postarticle"
            onClick={[Function]}
          >
            Post Article
          </a>
        </Link>
      </NavLink>
      <i
        className="fa fa-bell notnotif text-dark mt-3"
      />
      <NavLink
        aria-expanded="true"
        className="nav-link dropdown-toggle"
        data-target="#profileDropdown"
        data-toggle="dropdown"
        role="button"
        to="#null"
      >
        <Link
          aria-current={null}
          aria-expanded="true"
          className="nav-link dropdown-toggle"
          data-target="#profileDropdown"
          data-toggle="dropdown"
          role="button"
          to="#null"
        >
          <a
            aria-current={null}
            aria-expanded="true"
            className="nav-link dropdown-toggle"
            data-target="#profileDropdown"
            data-toggle="dropdown"
            href="/#null"
            onClick={[Function]}
            role="button"
          >
            <i
              className="fa fa-user-circle ml-3 mr-2 mt-2"
            />
            <span
              className="mr-1"
            />
          </a>
        </Link>
      </NavLink>
      <div
        className="dropdown-menu ml-5"
        id="profileDropdown"
      >
        <div
          className="dropdown-item"
        >
          <NavLink
            className="nav-link"
            to="/myProfile/"
          >
            <Link
              aria-current={null}
              className="nav-link"
              to="/myProfile/"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/myProfile/"
                onClick={[Function]}
              >
                Profile
              </a>
            </Link>
          </NavLink>
        </div>
        <div
          className="dropdown-item"
        >
          <NavLink
            className="nav-link"
            to="#null"
          >
            <Link
              aria-current={null}
              className="nav-link"
              to="#null"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/#null"
                onClick={[Function]}
              >
                Settings
              </a>
            </Link>
          </NavLink>
        </div>
        <div
          className="dropdown-item"
        >
          <NavLink
            className="nav-link"
            onClick={[Function]}
            to="/"
          >
            <Link
              aria-current="page"
              className="nav-link active"
              onClick={[Function]}
              style={Object {}}
              to="/"
            >
              <a
                aria-current="page"
                className="nav-link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Logout
              </a>
            </Link>
          </NavLink>
        </div>
      </div>
    </div>
  </SignedInLinks>
</Router>
`;
